libs/datasource/jwt/src/yjs/index.ts

```diff
@@ -324,53 +324,59 @@ export class YjsAdapter implements AsyncDatabaseAdapter<YjsContentOperation> {
                 saveAs(
                     new Blob([binary]),
-                    `affine_workspace_${new Date().toDateString()}.apk`
+                    `affine_workspace_${new Date().toDateString()}.affine`
                 );
             },
             load: async () => {
-                const handles = await window.showOpenFilePicker({
-                    types: [
-                        {
-                            description: 'AFFiNE Package',
-                            accept: {
-                                // eslint-disable-next-line @typescript-eslint/naming-convention
-                                'application/affine': ['.apk'],
+                try {
+                    const handles = await window.showOpenFilePicker({
+                        types: [
+                            {
+                                description: 'AFFiNE Package',
+                                accept: {
+                                    // eslint-disable-next-line @typescript-eslint/naming-convention
+                                    'application/affine': ['.affine'],
+                                },
                             },
-                        },
-                    ],
-                });
-                const [file] = (await fromEvent(handles)) as File[];
-                const binary = await file?.arrayBuffer();
-                // await this._provider.idb.clearData();
-                const doc = new Doc({ autoLoad: true, shouldLoad: true });
-                let updated = 0;
-                let isUpdated = false;
-                doc.on('update', () => {
-                    isUpdated = true;
-                    updated += 1;
-                });
-                setInterval(() => {
-                    if (updated > 0) {
-                        updated -= 1;
+                        ],
+                    });
+                    const [file] = (await fromEvent(handles)) as File[];
+                    const binary = await file?.arrayBuffer();
+                    // await this._provider.idb.clearData();
+                    const doc = new Doc({ autoLoad: true, shouldLoad: true });
+                    let updated = 0;
+                    let isUpdated = false;
+                    doc.on('update', () => {
+                        isUpdated = true;
+                        updated += 1;
+                    });
+                    setInterval(() => {
+                        if (updated > 0) {
+                            updated -= 1;
+                        }
+                    }, 500);
+
+                    const update_check = new Promise<void>(resolve => {
+                        const check = async () => {
+                            while (!isUpdated || updated > 0) {
+                                await sleep();
+                            }
+                            resolve();
+                        };
+                        check();
+                    });
```

Comment:
What scenarios need to listen to update

Reply:
i don't know, only add try.catch logic.
There is a problem with the previous logic: when click cancel btn, will throw error